### PR TITLE
4th Terraform Recipe - Kafka and Mirrormaker2

### PR DESCRIPTION
**Pull request by dewan-ahmed**
# What changed, and why it matters

Resolves https://github.com/aiven/devportal/issues/885

## Feedback requested:

- Please run the Terraform code and ensure that it works (I did multiple times and works from my end)
- I haven't used MirrorMaker before so please check if the descriptions of the flow makes sense
- I used *i1*, *i2*, *f1* naming which is not descriptive. Please let me know if I can improve the PR with a more descriptive naming 
- Please check if the Mermaid diagram makes sense
- Is the blog under the reference section appropriate to mention?
- Anything else that you find would improve this recipe

### CODE REVIEW

**Comment by dewan-ahmed** [comment]
From a quick glance, my assumption is that the following broken link is causing the fail:

```
(docs/products/opensearch/howto/sample-dataset: line  157) broken    https://opensearch.org/docs/latest/opensearch/rest-api/index-apis/create-index/#mappings - Anchor 'mappings' not found
(docs/products/opensearch/howto/sample-dataset: line  240) broken    https://opensearch.org/docs/latest/opensearch/rest-api/index-apis/create-index/#dynamic-mapping-types - Anchor 'dynamic-mapping-types' not found
```

I'll check it in detail during the week of June6th.

**Review by lornajane — CHANGES_REQUESTED** [review]
Sharing this initial review for now - as discussed, the terraform recipe didn't set up the replication flow I expected, so that also needs checking.

File: docs/tools/terraform/reference/cookbook/kafka-mirrormaker-recipe.rst [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+.. code:: bash
+
+   resource "aiven_kafka_mirrormaker" "mm" {
+    project      = data.aiven_project.kafka-mm-project1.project
+    cloud_name   = "google-europe-west1"
+    plan         = "startup-4"
+    service_name = "mm"
+
+    kafka_mirrormaker_user_config {
+      ip_filter = [
```
[the remark this answers is not in the document]
Comment (dewan-ahmed) [reply docs/tools/terraform/reference/cookbook/kafka-mirrormaker-recipe.rst:47]: Thanks for the suggestion. I'm looking for more info on this config option and will include a few lines for the reader.

[the remark this answers is not in the document]
Comment (dewan-ahmed) [reply docs/tools/terraform/reference/cookbook/kafka-mirrormaker-recipe.rst:50]: @lornajane I'm waiting to gather more info on this specific config option. Once I have this info, I'll add it to the document and mark this comment as 'Resolved'.

**Review by ftisiot — CHANGES_REQUESTED** [review]
Tried it, but it fails with the following when calling `terraform plan`

```
╷
│ Error: Reference to undeclared resource
│
│   on services.tf line 2, in resource "aiven_kafka_mirrormaker" "mm":
│    2:   project      = data.aiven_project.kafka-mm-project1.project
│
│ A data resource "aiven_project" "kafka-mm-project1" has not been declared in the root module.
╵
╷
│ Error: Reference to undeclared resource
│
│   on services.tf line 61, in resource "aiven_kafka" "source":
│   61:  project                 = data.aiven_project.kafka-mm-project1.project
│
│ A data resource "aiven_project" "kafka-mm-project1" has not been declared in the root module.
╵
╷
│ Error: Reference to undeclared resource
│
│   on services.tf line 86, in resource "aiven_kafka" "target":
│   86:  project                 = data.aiven_project.kafka-mm-project1.project
│
│ A data resource "aiven_project" "kafka-mm-project1" has not been declared in the root module.
```

**Comment by dewan-ahmed** [comment]
The recipes don't include the common files that are required for execution (for example, provider.tf, variables.tf, etc). But made a small change for the project name for consistency and added the following note:

.. Tip::

    Be sure to check out the :doc:`getting started guide <../../get-started>` to learn about the common files required to execute the following recipe. For example, you'll need to declare the variables for ``project_name`` and ``api_token``.

Please let me know if this helps you as a reader and I'll add this note to the other recipes as well.

**Comment by ftisiot** [comment]
Hi Dewan, I changed the recipe to be a bit more readable and added a section at the end about what to expect and how to understand if it works. Let me know what you think of it

**Comment by mhoffm-aiven** [comment]
I'm very much for making the manifests self-contained and including everything that it needs to run stand alone.

**Comment by dewan-ahmed** [comment]
> I'm very much for making the manifests self-contained and including everything that it needs to run stand alone.

Hi @ftisiot @mhoffm-aiven I'll be making the change next quarter via https://github.com/aiven/devportal/issues/959

**Comment by dewan-ahmed** [comment]
Hi @ftisiot I appreciate the changes/additions to this recipe. I reviewed them ✅

**Review by ftisiot — APPROVED** [review]
LGTM

**Pull request merged**